captura/views.py

```diff
@@ -410,6 +412,95 @@ def get_form_edit_integrante(request, id_integrante):
+    """ This view allows any user to create a new transaccion
+    regardless of it's type either ingreso or egreso.
+
+    """
+    if request.is_ajax() and request.method == 'POST':
+        response_data = {}
+        if request.POST['id_transaccion']:  # In case of updating
+            transaccion = get_object_or_404(Transaccion, pk=request.POST['id_transaccion'])
+            transaccion_form = TransaccionForm(request.POST, instance=transaccion)
+        else:  # In case of creation
+            transaccion_form = TransaccionForm(request.POST)
+        if transaccion_form.is_valid():
+            transaccion_form.save()
+            if transaccion_form.cleaned_data['es_ingreso']:
+                if hasattr(transaccion_form.instance, 'ingreso'):
+                    ingreso_form = IngresoForm(id_familia, request.POST,
+                                               instance=transaccion_form.instance.ingreso)
+                else:
+                    ingreso_form = IngresoForm(id_familia, request.POST)
+                if ingreso_form.is_valid():
+                    ingreso = ingreso_form.save(commit=False)
+                    ingreso.transaccion = transaccion_form.instance
+                    print(ingreso_form.errors)
+                    ingreso.save()
+                    response_data['msg'] = 'Ingreso guardado con éxito'
+                    return JsonResponse(response_data)
+                return HttpResponse(ingreso_form.errors.as_json(),
+                                    status=400,
+                                    content_type='application/json')
+            response_data['msg'] = 'Egreso guardado con exito'
+            return JsonResponse(response_data)
+        return HttpResponse(transaccion_form.errors.as_json(),
+                            status=400,
+                            content_type='application/json')
+    return HttpResponseBadRequest()
+
+
+@login_required
+@user_passes_test(is_capturista)
+def update_transaccion_modal(request, id_transaccion):
+    """ Returns a form that can be used to edit an existing
+    transaccion.
+    """
+    if request.is_ajax():
+        context = {}
+        transaccion = get_object_or_404(Transaccion, pk=id_transaccion)
+        id_familia = transaccion.familia.pk
+        context['id_familia'] = id_familia
+        id_transaccion = transaccion.pk
+        context['transaccion_form'] = TransaccionForm(instance=transaccion,
+                                                      initial={'id_transaccion': id_transaccion})
+        if hasattr(transaccion, 'ingreso'):
+            context['ingreso_form'] = IngresoForm(id_familia, instance=transaccion.ingreso)
+        return render(request, 'captura/edit_ingreso_egreso_form.html', context)
+    return HttpResponseBadRequest()
+
+
+@login_required
+@user_passes_test(is_capturista)
+def delete_transaccion(request):
+    pass
+
+
+@login_required
+@user_passes_test(is_capturista)
+def transacciones(request, id_familia):
```

Comment:
please change the name to something like 'list_transacciones' which gives more info and is more readable.